### PR TITLE
feat: reduce vmss cache refresh in parallel disk attach/detach

### PR DESCRIPTION
**Pull request by andyzhangx**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
feat: reduce vmss cache refresh in parallel disk attach/detach
This PR adds a lock in get vmss operation, in 1K disk attach/detach load test, it could reduce ~20%(67/338=20%) vmss cache refresh operations. It's not necessary to refresh cache every  time when nodeName is not found in vmss cache, especially in parallel workloads.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
/hold

#### Does this PR introduce a user-facing change?

```release-note
feat: reduce vmss cache refresh in parallel disk attach/detach
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
feat: reduce vmss cache refresh in parallel disk attach/detach
```

### CODE REVIEW

**Comment by feiskyer** [comment]
>  It's not necessary to refresh cache every time when nodeName is not found in vmss cache, especially in parallel workloads.

If the new nodes are created continuously by autoscaler, then the cache needs to be refreshed. or else, newly joined nodes may be deleted.

File: pkg/provider/azure_vmss.go [hunk 1]
```diff
@@ -76,6 +76,8 @@ type ScaleSet struct {
 	vmssCache                 *azcache.TimedCache
 	vmssVMCache               *sync.Map // [resourcegroup/vmssname]*azcache.TimedCache
 	availabilitySetNodesCache *azcache.TimedCache
+	// lockMap in cache refresh
+	lockMap *lockMap
```
Comment (feiskyer) [inline pkg/provider/azure_vmss.go:80]: how is this map set and refreshed?

Comment (andyzhangx) [reply pkg/provider/azure_vmss.go:80]: it's in L98:
```
lockMap:         newLockMap(),
```

**Comment by andyzhangx** [comment]
> > It's not necessary to refresh cache every time when nodeName is not found in vmss cache, especially in parallel workloads.
> 
> If the new nodes are created continuously by autoscaler, then the cache needs to be refreshed. or else, newly joined nodes may be deleted.

@feiskyer this PR does not change the original logic, this PR make vmss list operation as serial by using lock, and then it would take second look in vmss cache, that would improve the cache hit ratio when there are large mount of vmss list operations in parallel.

**Review by feiskyer — APPROVED** [review]
/lgtm

**Comment by feiskyer** [comment]
> @feiskyer this PR does not change the original logic, this PR make vmss list operation as serial by using lock, and then it would take second look in vmss cache, that would improve the cache hit ratio when there are large mount of vmss list operations in parallel.

Reasonable. LGTM.

**Pull request merged**